### PR TITLE
Expose BackgroundService.ExecuteTask

### LINKED ISSUE #35991: BackgroundService TaskStatus
### Is your feature request related to a problem? Please describe. 
As a developer I'm unable able to determine if a registered BackgroundService is running and/or has ran to competition or faulted.

### Describe the solution you'd like
The based abstract class should have a TaskStatus property added which shows the status of the ExecuteAsync task.

## Proposed API

```diff
namespace Microsoft.Extensions.Hosting
{
    /// <summary>
    /// Base class for implementing a long running <see cref="IHostedService"/>.
    /// </summary>
    public abstract class BackgroundService
    {
+        /// <summary>
+        /// Gets the Task that is executing the background operation.
+        /// </summary>
+        public Task ExecutingTask;
    }
}
```

## Usage example

```C#
BackgroundService s = ...;
s.ExecutingTask.ContinueWith(t =>
{
    if (t.IsFaulted)
    {
        Console.WriteLine(t.Exception.ToString());
    }
});
```

### LINKED ISSUE #36017: When awaiting an async method from ExecuteAsync which throws an unhandled exception, the service hangs
### Describe the bug
In a .Net Core Worker Service, whenever an exception is thrown inside an async method, and you happened to not surround the method invocation in a try/catch, the application will simply hang when it hits the exception -- giving no sign that an exception has occured at all.

The same code applied within a simple Console Application does indeed work as expected, and you get the normal "unhandled exception" behaviour.

### To Reproduce
Run sample worker service available at https://github.com/hognevevle/WorkerServicePlayground

This project was created using the Worker Service project template, and my only changes are done inside `Worker.cs`.

### Expected behavior
The exception should be propagated up to the caller, and not simply disappear.

### Screenshots
https://drive.google.com/file/d/1VCfrJY_a45gJ17uQXwGAq_krc8n-iteB/view?usp=drivesdk

### Additional context
```dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.100
 Commit:    cd82f021f4

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18362
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.100\

Host (useful for support):
  Version: 3.1.0
  Commit:  65f04fb6db

.NET Core SDKs installed:
  3.0.101 [C:\Program Files\dotnet\sdk]
  3.1.100 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.14 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.14 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download```

### PR DESCRIPTION
**Pull request by eerhardt**
Expose the task that executes the background service, so consumers can check if it is running and/or has ran to competition or faulted.

Use the new ExecuteTask to log exceptions when a BackgroundService fails after await, instead of appearing to hang.

Fix #35991
Fix #36017

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs [hunk 1]
```diff
@@ -52,6 +52,14 @@ public async Task StartAsync(CancellationToken cancellationToken = default)
             {
                 // Fire IHostedService.Start
                 await hostedService.StartAsync(combinedCancellationToken).ConfigureAwait(false);
+
+                if (hostedService is BackgroundService backgroundService)
+                {
+                    _ = backgroundService.ExecuteTask.ContinueWith(t =>
```
Comment (davidfowl) [inline src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs:58]: @stephentoub might disagree with me but I'd prefer a local async function than continue with:

```C#
async Task HandleException()
{
    try
    {
        await backgroundService.ExecuteTask;
    }
    catch (Exception ex)
    {
        _logger.BackgroundServiceFaulted(t.Exception);
    }
}
```

Comment (stephentoub) [reply src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs:58]: No disagreement.  In general I'd suggest using async as well unless profiling demonstrates there's an advantage to ContinueWith (there can be but the situations are pretty limited).  Especially if this isn't a very frequent operation, async/await all the way.

**Review by maryamariyan — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs (hunk 1 above)
Comment (eerhardt) [reply src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs]: Fixed.

**Comment by eerhardt** [comment]
@davidfowl - any more thoughts on this? Or is it good to merge?

**Review by davidfowl — APPROVED** [review]

**Pull request merged**